### PR TITLE
Send os and arch during long poll

### PR DESCRIPTION
**Pull request by luketomlinson**
We want the runner to self report it's os/architecture. This allows us to make certain decisions about upgrades, etc.

### CODE REVIEW

File: src/Runner.Common/RunnerDotcomServer.cs [hunk 1]
```diff
@@ -163,6 +164,8 @@ private async Task<DistributedTask.WebApi.Runner> AddOrReplaceRunner(int runnerG
                         {"ephemeral", agent.Ephemeral},
                         {"labels", agent.Labels},
                         {"public_key", publicKey},
+                        {"os", VarUtil.OS},
```
Comment (TingluoHuang) [inline src/Runner.Common/RunnerDotcomServer.cs:167]: This or the combined package name, ex Linux-x64

**Review by TingluoHuang — APPROVED** [review]

File: src/Runner.Common/RunnerDotcomServer.cs (hunk 1 above)
Comment (luketomlinson) [reply src/Runner.Common/RunnerDotcomServer.cs:167]: We send back just the `os` in the REST API, so it's easy enough to concatenate them if we need to. https://docs.github.com/en/rest/actions/self-hosted-runners?apiVersion=2022-11-28

**Review by TingluoHuang — DISMISSED** [review]

File: src/Sdk/DTGenerated/Generated/TaskAgentHttpClientBase.cs [hunk 2]
```diff
@@ -483,6 +485,16 @@ public TaskAgentHttpClientBase(Uri baseUrl, HttpMessageHandler pipeline, bool di
                 queryParams.Add("runnerVersion", runnerVersion);
             }
 
+            if (os != null)
+            {
+                queryParams.Add("os", os);
```
Comment (TingluoHuang) [inline src/Sdk/DTGenerated/Generated/TaskAgentHttpClientBase.cs:490]: make sure you double check configure the runner with your change with github.com
I am 99.9% sure the extra query param will just no-opt on the service side.

**Review by TingluoHuang — APPROVED** [review]

File: src/Sdk/DTGenerated/Generated/TaskAgentHttpClientBase.cs (hunk 2 above)
Comment (luketomlinson) [reply src/Sdk/DTGenerated/Generated/TaskAgentHttpClientBase.cs:490]: done

**Pull request merged**